sql/core/src/main/scala/org/apache/spark/sql/sources/interfaces.scala

```diff
@@ -195,6 +196,8 @@ abstract class BaseRelation {
    *  java.lang.String -> UTF8String
    *  java.lang.Decimal -> Decimal
    *
+   * If `needConversion` is `false`, buildScan() should return an [[RDD]] of [[InternalRow]]
```

Comment:
If we move to let InternalRow no inherit from Row, maybe we could introduce another `internalBuildScan` for internal data sources, which will return `RDD[InternalRow]`, had not figured out all the details yet.